tests/Progress/ProgressHashesTest.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace BristolSU\Support\Tests\Progress;
+
+use BristolSU\Support\Progress\ProgressHashes;
+use BristolSU\Support\Tests\TestCase;
+use Illuminate\Foundation\Testing\WithFaker;
+
+class ProgressHashesTest extends TestCase
+{
+    use WithFaker;
+
+    protected $Model;
+    protected $Table;
+
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        $this->Model = (new ProgressHashes);
+        $this->Table = $this->Model->getTable();
+    }
+
+    public function getModelData()
+    {
+        return [
+            'item_key' => $this->faker('en_UK')->slug,
```

Comment:
Could you hardcode in values rather than use faker? I'm never convinced about using Faker in tests like this where you're using the faked values - if a test fails when e.g. the item key starts with a z (random example), then the test will become unreliable in its results as it'll only sometimes fail.

I think a better way is to hardcode in a value for the item key (or even better, a data provider to test lots of values). That way, if a test fails you know it'll always fail and won't depend on a random string, and it's more verbose what's going on (using tests as documentation, you can see exactly what value item_key should take).

Faker is obviously great if you just need a dummy value to create a model or something, but when actually testing using faked values it may be a bit too risky.

Very happy to have a discussion about this - I know you prefer using Faker and have mentioned that for Atlas  so it'd be good to work out where to use each method!